includes/class-wc-payments-upe-checkout.php

```diff
@@ -105,6 +111,12 @@ public function get_payment_fields_js_config() {
 		$payment_fields['wcBlocksUPEAppearance']    = get_transient( UPE_Payment_Gateway::WC_BLOCKS_UPE_APPEARANCE_TRANSIENT );
 		$payment_fields['cartContainsSubscription'] = $this->gateway->is_subscription_item_in_cart();
 
+		if ( WC_Payments_Features::is_upe_legacy_enabled() ) {
+			$payment_fields['checkoutTitle']        = $this->gateway->get_checkout_title();
+			$payment_fields['upePaymentIntentData'] = $this->gateway->get_payment_intent_data_from_session();
+			$payment_fields['upeSetupIntentData']   = $this->gateway->get_setup_intent_data_from_session();
+		}
+
```

Comment:
If the split UPE is enabled, then `$this->gateway` will be an instance of `UPE_Split_Payment_Gateway`, whereas if the split UPE is not enabled, it will be an instance of `UPE_Payment_Gateway` (see `class-wc-payments.php` lines 394 & 398). In both the legacy implementation and the split UPE implementation, the `WC_Payments_UPE_Checkout` class is initialised (see `class-wc-payments.php` lines 396 & 399). This is done to hydrate the `wcpay_upe_config` or `wcpay_config` JS variable, which will be used by either the legacy or split UPE front-end script (see `class-wc-payments.php` line 237). 

These variables are only needed when the legacy UPE gateway is present and they have pretty irrelevant values when the split UPE is enabled, so I decided not to include them in the config when the split UPE is enabled. 

Not sure if I completely understood your question, so let me know if I missed the crux of your suggestion here!